### PR TITLE
Android - allow changing min sdk version

### PR DESCRIPTION
**Pull request by PieterVanPoyer**
### Platforms affected

Android

### Motivation and Context

- Fixes https://github.com/apache/cordova-android/issues/1070 

Give the users the opportunity to overwrite the minSdkVersion without a failing build.

> Manifest merger failed : uses-sdk:minSdkVersion 19 cannot be smaller than version 22 declared in
> library [:CordovaLib]

### Description

This PR uses the overriden cdvMinSdkVersion if it is provided, else it is defaulted to 22 for now.

### Testing

I did ran the Java Android tests.
I tested it in next repo: https://github.com/PieterVanPoyer/cordova-plugin-splashscreen-testing-app/tree/testing/changing-min-sdk-version 

### Checklist

- [x] I've run the tests to see all new and existing tests pass
- [ ] I added automated test coverage as appropriate for this change
- [x] Commit is prefixed with `(platform)` if this change only applies to one platform (e.g. `(android)`)
- [x] If this Pull Request resolves an issue, I linked to the issue in the text above (and used the correct [keyword to close issues using keywords](https://help.github.com/articles/closing-issues-using-keywords/))
- [ ] I've updated the documentation if necessary

### CODE REVIEW

File: framework/build.gradle
Comment (breautek) [inline framework/build.gradle:28]:
```suggestion
```
I don't think we need to print output in this block case.

File: framework/build.gradle
Comment (breautek) [inline framework/build.gradle:25]:
```suggestion
        println "[Cordova] cdvMinSdkVersion is overridden, try it at your own risk."
```

"overridden" typo.

Additionally, I think we should ensure that cdvMinSdkVersion matches case to avoid confusion. These variables can be provided by the user via [these methods](https://cordova.apache.org/docs/en/latest/guide/platforms/android/index.html#setting-gradle-properties).

**Review by breautek — COMMENTED** [review]
Just a few nitpicks.

File: framework/build.gradle
Comment (PieterVanPoyer) [reply framework/build.gradle]: Absolutely correct.

**Review by breautek — APPROVED** [review]
lgtm

**Comment by PieterVanPoyer** [comment]
@breautek what do you think? Should we wait for another review? Or can this be merged now, or should this be merged later? Kind regards

File: framework/build.gradle
Comment (timbru31) [inline framework/build.gradle:23]: Shouldn't we guard it with a check that the given string is indeed an integer?

File: framework/build.gradle
Comment (timbru31) [inline framework/build.gradle:25]: The log statement should use single quotes.

File: framework/build.gradle
Comment (timbru31) [inline framework/build.gradle:24]: AFAIK Gradle's syntax is simply `cdvMinSdkVersion as int`

**Review by timbru31 — CHANGES_REQUESTED** [review]

File: framework/build.gradle
Comment (PieterVanPoyer) [reply framework/build.gradle]: I added an isInteger check: https://github.com/apache/cordova-android/pull/1117/files#diff-3dc11ca3a71905f437e1e170f963e9854da27943efcff23bf87bfe87abeda513R23

But, I didn't see any other guards in the cordova gradle files. I might have missed it off course as I am no Gradle expert. 
IMO, if we want to assist the users with additional type checking, it should be in a seperate routine, and the wrong types must be logged with warning or error to the user (if that is possible).

**Comment by PieterVanPoyer** [comment]
@timbru31 I completed/resolved your requesting changes. I did learn a bit about Groovy and Gradle with this PR. Nice. If there are still remarks, just let it know.

**Review by timbru31 — APPROVED** [review]
LGTM; thanks for addressing my comments.

**Comment by breautek** [comment]
I'll give it a couple more days or so to give a chance for anybody else to pitch in. Assuming there are no objections I'll merge.

File: framework/build.gradle
Comment (PieterVanPoyer) [reply framework/build.gradle]: Changed it to single quotes. Easy. 
There are although other places printing with double quotes. https://github.com/apache/cordova-android/blob/master/framework/cordova.gradle#L53

File: framework/build.gradle
Comment (PieterVanPoyer) [reply framework/build.gradle]: Yes nice. I think it is time for me, to pick up a Gradle/Groovy tutorial.

**Review by erisu — APPROVED** [review]

**Comment by breautek** [comment]
Tests are green, and there haven't been any objections so I'm merging!

Thank you @PieterVanPoyer for your time and effort in preparing this PR and thank you @erisu and @timbru31 for your reviews.

**Pull request merged**
